storage/innobase/xtrabackup/src/redo_log.cc

```diff
@@ -953,17 +954,31 @@ void Archived_Redo_Log_Monitor::thread_func() {
     std::string dir;
     std::string subdir;
   } archive;
-
+  
   read_mysql_variables(mysql, "SHOW VARIABLES", vars, true);
 
-  if (redo_log_archive_dirs == nullptr || *redo_log_archive_dirs == 0) {
+  if (!(redo_log_archive_dirs == nullptr || *redo_log_archive_dirs == 0)) {
+    xb::info() << "innodb_redo_log_archive_dirs is already  set .";
+    innodb_redo_log_archive = true;
```

Comment:
what the code is doing and what the description of the variable says is contradicting.
You set this to true if the redo archive is already set on the server. Based on your comment this should be true when PXB changed redo archive dir from null to xtrabackup_redo_log_arch_dir.

Reply:
I would rather re-work this whole block. Its a bit counter intuitive. Some thing like below will be better:

```
if (redo_log_archive_dirs == nullptr || *redo_log_archive_dirs == 0) {
  std::ostringstream query;
  if (xtrabackup_redo_log_arch_dir) {
    query << "SET GLOBAL innodb_redo_log_archive_dirs ="
          << "\"" << xtrabackup_redo_log_arch_dir << "\"";
    xb_mysql_query(mysql, query.str().c_str(), false, true); <-- we need to keep this to fail in case of error
    innodb_redo_log_archive = true;
  } else {
    xb::info() << "Redo Log Archiving is not set up.";
    free_mysql_variables(vars);
    mysql_close(mysql);
    my_thread_end();
    return;
  }
}
xb::info() << "xtrabackup redo_log_arch_dir is now  set .";
```